umh-core/pkg/cpuhealth/snapshot_test.go

```
@@ -0,0 +1,213 @@
+			mutate(fs)
+		}
+		return cpuhealth.NewCgroupSampler(fs, base)
+	}
+
+	It("stamps one timestamp, builds the environment from two facts, and fails the whole snapshot only on cpu.stat", func() {
+		// Spec 1 + 2: a real Read builds every field off one read, and the
+		// snapshot is constructible from OUTSIDE pkg/diagnosis through Known and
+		// Unknown. The single Timestamp is set once per tick.
+		ctx := context.Background()
+		smp, err := sampler(nil).Read(ctx)
+		Expect(err).To(BeNil())
```

golangci-lint (2.12.2)

[error] ginkgo-linter: wrong error assertion. Consider using `Expect(err).ToNot(HaveOccurred())` instead

(ginkgolinter)